script.js

```diff
@@ -1,4 +1,216 @@
+// Project 1: Scissors Paper Stone
+
+// Define constants to use in multiple functions to avoid bugs due to typos.
+var SCISSORS = "scissors";
+var PAPER = "paper";
+var STONE = "stone";
+var numRolls = 0;
+var numWins = 0;
+var winPercent = "";
+var winRecord = "";
+var userName = "";
+var gameMode = "waiting for username";
+var mostRecentWinner = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == "waiting for username") {
+    userName = input;
+    gameMode = "normal";
+    return `Hello ${userName}! You can start playing by entering "scissors", "paper" or "stone". `;
+  } else {
+    if (gameMode == "normal") {
+      return spsGame(userName, input);
+    }
+    if (gameMode == "reverse") {
+      return reversedSPSGame(userName, input);
+    }
+    if (gameMode == "korean") {
+      return koreanSPSGame(userName, input);
+    }
+  }
+};
```

Comment:
Really good use of conditional statements as well as the helper functions. The main function is very clean and logical. Really good job with this!